common/Message.hpp

```diff
@@ -32,32 +33,9 @@ class Message
 
     /// Construct a text message.
     /// message must include the full first-line.
-    Message(const std::string& message,
-            const enum Dir dir) :
-        _forwardToken(getForwardToken(message.data(), message.size())),
-        _data(copyDataAfterOffset(message.data(), message.size(), _forwardToken.size())),
-        _tokens(StringVector::tokenize(_data.data(), _data.size())),
-        _id(makeId(dir)),
-        _type(detectType()),
-        _hash(0)
-    {
-        LOG_TRC("Message " << abbr());
-    }
-
-    /// Construct a message from a string with type and
-    /// reserve extra space (total, including message).
-    /// message must include the full first-line.
-    Message(const std::string& message,
-            const enum Dir dir,
-            const size_t reserve) :
-        _forwardToken(getForwardToken(message.data(), message.size())),
-        _data(copyDataAfterOffset(message.data(), message.size(), _forwardToken.size())),
-        _tokens(StringVector::tokenize(message.data() + _forwardToken.size(), message.size() - _forwardToken.size())),
-        _id(makeId(dir)),
-        _type(detectType()),
-        _hash(0)
+    Message(const std::string_view message, const enum Dir dir)
+        : Message(message.data(), message.size(), dir)
     {
-        _data.reserve(std::max(reserve, message.size()));
         LOG_TRC("Message " << abbr());
```

Comment:
The ctor that this variant forwards to also has "LOG_TRC("Message " << abbr());" so I presume we end up with duplicate "LOG_TRC" if this ctor is used. We probably don't want that?

Reply:
Good point.